app/assets/javascripts/locales/i18n.js

```diff
@@ -1,380 +1,2 @@
-
-  number = this.toNumber(size, options);
-  number = options.format.replace("%u", unit).replace("%n", number);
-
-  return number;
-};
-
-I18n.listJoiner = function (listOfStrings, delimiter) {
-  if (listOfStrings.length === 1) {
-    return listOfStrings[0];
-  }
-
-  if (listOfStrings.length === 2) {
-    return listOfStrings[0] + " " + delimiter + " " + listOfStrings[1];
-  }
-
-  var lastString = listOfStrings.pop();
-  return listOfStrings.concat(delimiter).join(`, `) + " " + lastString;
-};
-
-I18n.pluralizer = function (locale) {
-  var pluralizer = this.pluralizationRules[locale];
-  if (pluralizer !== undefined) return pluralizer;
-  return this.pluralizationRules["en"];
-};
-
-I18n.findAndTranslateValidNode = function (keys, translation) {
-  for (var i = 0; i < keys.length; i++) {
-    var key = keys[i];
-    if (this.isValidNode(translation, key)) return translation[key];
-  }
-  return null;
-};
-
-I18n.pluralize = function (translation, scope, options) {
-  if (typeof translation !== "object") return translation;
-
-  options = this.prepareOptions(options);
-  var count = options.count.toString();
-
-  var pluralizer = this.pluralizer(options.locale || this.currentLocale());
-  var key = pluralizer(Math.abs(count));
-  var keys = typeof key === "object" && key instanceof Array ? key : [key];
-
-  var message = this.findAndTranslateValidNode(keys, translation);
-
-  if (message !== null || options.ignoreMissing) {
-    return message;
-  }
-
-  return this.missingTranslation(scope, keys[0]);
-};
-
-I18n.missingTranslation = function (scope, key, options) {
-  var message = "[" + this.currentLocale() + this.SEPARATOR + scope;
-
-  if (key) {
-    message += this.SEPARATOR + key;
-  }
-
-  if (options && options.hasOwnProperty("count")) {
-    message += " count=" + JSON.stringify(options.count);
-  }
-
-  return message + "]";
-};
-
-I18n.currentLocale = function () {
-  return I18n.locale || I18n.defaultLocale;
-};
-
-I18n.enableVerboseLocalization = function () {
-  var counter = 0;
-  var keys = {};
-  var t = I18n.t;
-
-  I18n.noFallbacks = true;
-
-  I18n.t = I18n.translate = function (scope, value) {
-    var current = keys[scope];
-    if (!current) {
-      current = keys[scope] = ++counter;
-      var message = "Translation #" + current + ": " + scope;
-      if (value && Object.keys(value).length > 0) {
-        message += ", parameters: " + JSON.stringify(value);
-      }
-      // eslint-disable-next-line no-console
-      console.info(message);
-    }
-    return t.apply(I18n, [scope, value]) + " (#" + current + ")";
-  };
-};
-
-I18n.enableVerboseLocalizationSession = function () {
-  sessionStorage.setItem("verbose_localization", "true");
-  I18n.enableVerboseLocalization();
-
-  return "Verbose localization is enabled. Close the browser tab to turn it off. Reload the page to see the translation keys.";
-};
-
-// shortcuts
-I18n.t = I18n.translate;
+require("discourse/loader-shims");
+require("discourse-i18n");
```

Comment:
This is a bit of a hack to ensure `globalThis.I18n` is defined before the rest of the locale code runs. It works for now, but it's probably better to make the locale bundle lazier – turn the locale code into an AMD module with a dependency on `discourse-i18n`, rather than running synchronously and expects the `I18n` global to already be defined as the script is loaded.